src/transform-namespaces.test.ts.md

```diff
@@ -0,0 +1,143 @@
+
+## add namespace
+
+> Snapshot 1
+
+    `// SPDX-License-Identifier: UNLICENSED␊
+    pragma solidity ^0.8.20;␊
+    ␊
+    contract C1 {␊
+        function __C1_init() internal onlyInitializing {␊
+        }␊
+    ␊
+        function __C1_init_unchained() internal onlyInitializing {␊
+        }␊
+    }␊
+    ␊
+    contract C2 {␊
+        function __C2_init() internal onlyInitializing {␊
+            __C2_init_unchained();␊
+        }␊
+    ␊
+        function __C2_init_unchained() internal onlyInitializing {␊
+            C2Storage storage $ = _getC2Storage();␊
+            $.s1 = "";␊
+        }␊
+        event B();␊
+    ␊
+        struct A { uint z; }␊
+    ␊
+        uint constant C = 3;␊
+        // b␊
+        /// @custom:oz-upgrades-unsafe-allow state-variable-immutable␊
+        uint immutable y = 2;␊
+    ␊
+        /// @custom:storage-location erc7201:openzeppelin.storage.C2␊
+        struct C2Storage {␊
+            // a␊
+            uint x;␊
+            uint z;␊
+    ␊
+            string s1;␊
+        }␊
+    ␊
+        // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.C2")) - 1))␊
+        bytes32 private constant C2StorageLocation = 0xf05a05e0e3d15983ea921cad031aaea3040e9d631039045748753d29c5d248f9;␊
+    ␊
+        function _getC2Storage() private pure returns (C2Storage storage $) {␊
+            assembly {␊
+                $.slot := C2StorageLocation␊
+            }␊
+        }␊
+    ␊
+        function f() public {␊
+            C2Storage storage $ = _getC2Storage();␊
+            $.z = 3;␊
+        }␊
+    ␊
+        function g() public {␊
+        }␊
+    }␊
+    ␊
+    contract C3 {␊
+        function __C3_init() internal onlyInitializing {␊
+        }␊
+    ␊
+        function __C3_init_unchained() internal onlyInitializing {␊
+        }␊
+        /// @custom:storage-location erc7201:openzeppelin.storage.C3␊
+        struct C3Storage {␊
+            address x;␊
+        }␊
+    ␊
+        // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.C3")) - 1))␊
+        bytes32 private constant C3StorageLocation = 0xaa7e0685867d809c517036b8f21e99d58bd04b1da2b202f167c355fdf82b4a94;␊
+    ␊
+        function _getC3Storage() private pure returns (C3Storage storage $) {␊
+            assembly {␊
+                $.slot := C3StorageLocation␊
+            }␊
+        }␊
+    }␊
+    ␊
+    contract C4 {␊
+        /// @custom:storage-location erc7201:openzeppelin.storage.C4␊
+        struct C4Storage {␊
+            address x;␊
+        }␊
+    ␊
+        // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.C4")) - 1))␊
+        bytes32 private constant C4StorageLocation = 0x536a56e760d844b098efcc16711808e0b18bad9e07c2e82c78312ab719318df8;␊
+    ␊
+        function _getC4Storage() private pure returns (C4Storage storage $) {␊
+            assembly {␊
+                $.slot := C4StorageLocation␊
+            }␊
+        }␊
+        constructor() {function __C4_init() internal onlyInitializing {␊
```

Comment:
This is expected because the test is not applying all the transformations. But it's very confusing so I will add the missing transformation to fix the constructor part.